### DISCUSSION TITLE
Tests with lng: cimode and i18n.exists

### DISCUSSION THREAD

**Discussion by otaciliolacerda** [i18next/i18next#1559] [Q&A]
I am loading i18n on my test setup like:
```js
i18n.use(initReactI18next).init({ lng: 'cimode' });
```

In my project I have:
```js
if (!i18n.exists(key)) {
      key = defaultKey;
}
```

Using this configuration, the `exists` function will always return false. 
When I set the resources property in the init configuration I am getting the translated text as output instead of the keys even with `lng: 'cimode'`. Am I doing something wrong? Is there a way to configure i18n on my tests in a way that `i18n.exist` can check if the key exists and I still get the translation keys instead of translations?

I am using:
```json
"i18next": "^19.8.9",
"react-i18next": "^11.8.7",
```

**Comment by jamuhl** [comment]
https://github.com/i18next/i18next/blob/master/src/Translator.js#L102 can you make a reproducible sample were you get the translated output when using cimode?

 https://jsfiddle.net/pkmutqc9/ <-- seems to work

**Reply by otaciliolacerda** [reply]
Hi @jamuhl thanks for the quick reply.

Maybe I got it wrong but on your example if I switch to English I see the translation text instead of the translation key. What I was expecting was the key for all cases. 

EDIT:

Sorry, I forgot to give an important information. It was working before I upgrade the dependencies. It was working for:
```json
"i18next": "^19.8.7",
"react-i18next": "^11.8.6",
```

**Reply by jamuhl** [reply]
can you try with latest? not sure - but might be only 19.8.7 had this...?

if language is cimode -> keys should be shown -> every other case a value

**Comment by otaciliolacerda** [comment]
Hi @jamuhl I am currently using the latest version. Let me try to explain the problem again.

If I set `lng: 'cimode'` I get the keys as expected but on my code `i18next.exists` function always return false and `t([key, key2, lastKey])` always returns the last key in the array.

I initially thought the `cimode` would change just the output but it is actually a "language". It skips the resolution process completely. Now I understand why when I click in the "English" button on your example it will remove the `cimode` and start to resolve the translation to that language.

It seems there is no way to use `cimode` to test what I want (with translation keys). I need to use the translation values to do it.

Thank you for the help and examples.
